tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -19,15 +19,17 @@ internal class TraceContext
 
         private readonly DateTimeOffset _utcStart = DateTimeOffset.UtcNow;
         private readonly long _timestamp = Stopwatch.GetTimestamp();
+        private readonly AzureAppServices _azureAppServicesContext;
```

Comment:
While I completely agree with the sentiment RE testing, I think adding this field is responsible for the increased allocations in the benchmarks.

You can hack around it a bit for testing using the approach we do when we need to test static methods on Tracer:
1. Create an analogue of `[TracerInstanceTestCollection]` e.g. `[AzureAppServiceTestCollection]`
2. Create an analogue of `[TracerRestorer]` which saves the static `AzureAppServices.Metadata` value and restores it afterwards, so you can set it in the test

Reply:
You're killing me, 8B for a reference. This code is so much more beautiful, it's worth it :p But yeah, ok.
Can we go for option 3? I leave the reference null if not in tests, and test for nullity when I want to use it in `DecorateRootSpan`?

Reply:
Fixed in 381e28c2aea503241bb53845989c93272d3cfb48 btw